### PR TITLE
Flamer Instant Ignition + Special Duration Scaling + Other Related Adjustments

### PR DESCRIPTION
**Pull request by ModdedMarionette**

### CODE REVIEW

**Comment by ModdedMarionette** [comment]
- Added back Flamer's instant ignition from friendly fire.
- Added special scaling for the burn duration of Flamer's friendly fire ignition.

**Comment by ModdedMarionette** [comment]
Made this a draft because my algorithm isn't working as well as I hoped.
Will open again once it's satisfactory.

File: src/game/server/swarm/asw_marine.cpp [hunk 1]
```diff
@@ -5063,6 +5063,12 @@ void CASW_Marine::ASW_Ignite( float flFlameLifetime, float flSize, CBaseEntity *
 
 	bool bFriendlyFire = IRelationType( pAttacker ) == D_LI;
 
+	// special scaling for flamer ff
+	if (asw_marine_ff_absorption.GetInt() > 0 && bFriendlyFire && pDamagingWeapon->Classify() == CLASS_ASW_FLAMER)
+	{
+		flFlameLifetime = MAX( 1.0f, flFlameLifetime * m_fFriendlyFireAbsorptionTime );
```
Comment (BenLubar) [inline src/game/server/swarm/asw_marine.cpp]: I recommend making the 1.0 a convar so it's easier to change the minimum burn time for testing and for challenges.

**Comment by ModdedMarionette** [comment]
@BenLubar 
I've made some adjustments here and there. Do verify if they are safe changes.
Otherwise this request is now ready to be merged.

File: src/game/server/swarm/asw_marine.cpp [hunk 2]
```diff
@@ -1433,7 +1433,7 @@ int CASW_Marine::OnTakeDamage_Alive( const CTakeDamageInfo &info )
 			}
 
 			// drop the damage down by our absorption buffer
-			bool bFlamerDot = !!(newInfo.GetDamageType() & ( DMG_BURN | DMG_DIRECT ) );
+			bool bFlamerDot = !!(newInfo.GetDamageType() & DMG_DIRECT);
```
Comment (BenLubar) [inline src/game/server/swarm/asw_marine.cpp:1436]: oh, good catch

**Pull request merged**